<launch>

  <!-- ======================= Launchers depended on ======================= -->

  <!--State manager -->

  <include file="$(find state_manager)/launch/stateManager.launch" />


  <!-- ============================= Namespace ============================= -->

  <arg name="namespace" default="pandora_vision/hole_detector" />


  <!-- ========================= Synchronizer node ========================= -->

  <node
    name="synchronizer_node"
    type="synchronizer_node"
    pkg="pandora_vision_hole_detector"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole_detector)/cfg/synchronizer_node_topics.yaml"
              command="load" />
  </node>


  <!-- ============================ Depth node ============================= -->

  <node
    name="depth_node"
    type="depth_node"
    pkg="pandora_vision_hole_detector"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole_detector)/cfg/depth_node_topics.yaml"
              command="load" />
  </node>


  <!-- ============================= RGB node ============================== -->

  <node
    name="rgb_node"
    type="rgb_node"
    pkg="pandora_vision_hole_detector"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole_detector)/cfg/rgb_node_topics.yaml"
              command="load" />
  </node>


  <!-- ========================= Hole Fusion node ========================== -->

  <node
    name="hole_fusion_node"
    type="hole_fusion_node"
    pkg="pandora_vision_hole_detector"
    respawn="False"
    output="screen"
    ns="$(arg namespace)">

    <rosparam file="$(find pandora_vision_hole_detector)/cfg/hole_fusion_node_topics.yaml"
              command="load" />
  </node>


  <node pkg="pandora_testing_tools" type="bag_player_for_tests" name="player"
    args="--bags $(find pandora_vision_hole_detector)/test/functional/simple_pcl_frame.bag">
      <remap from="/camera/depth_registered/points" to="/kinect/depth_registered/points" />
  </node>

  <test test-name="HoleDetectorTest" pkg="pandora_vision_hole_detector" type="hole_detector_test.py" />

</launch>
